base/commandLine.cpp

```diff
@@ -798,20 +815,139 @@ static bool autoDetect(Common::String path) {
 	GameList candidates(EngineMan.detectGames(files));
 	if (candidates.empty()) {
 		printf("ScummVM could not find any game in %s\n", path.c_str());
-		return false;
 	}
+	return candidates;
+}
+
+/** Display all games in the given directory, add it to config according to input */
+static bool detectGames(Common::String path, bool addToConfig) {
+	GameList candidates = getGameList(path);
+	if (candidates.empty())
+		return false;
 
 	// Print all the candidate found
 	printf("ID                   Description\n");
 	printf("-------------------- ---------------------------------------------------------\n");
 	for (GameList::iterator v = candidates.begin(); v != candidates.end(); ++v) {
 		printf("%-20s %s\n", v->gameid().c_str(), v->description().c_str());
 	}
-	// Set the active domain to the first one to start it.
-	ConfMan.setActiveDomain(candidates.begin()->gameid());
+
+	if (addToConfig) {
+		Common::String domain = candidates[0].preferredtarget();
+		ConfMan.setActiveDomain(domain);
+	}
 	return true;
 }
 
+/** Add one of the games in the given directory, or current directory if empty */
+static bool addGame(Common::String path) {
+	GameList candidates = getGameList(path);
+	if (candidates.empty())
+		return false;
+
+	int idx = 0;
+	// Pick one if there are several games
+	if (candidates.size() > 1) {
+		// Print game list
+		printf("Several games are detected. Please pick one game to add: \n");
+		int i = 1;
+		for (GameList::iterator v = candidates.begin(); v != candidates.end(); ++i, ++v) {
+			printf("%2i. %s\n", i, v->description().c_str());
```

Comment:
Maybe print the gameid as well, for consistency with detectGames(), and because some users are used to those.